lightningd/bitcoind.c

```diff
@@ -38,6 +38,43 @@ static const char *methods[] = {"getchaininfo", "getrawblockbyheight",
                                 "sendrawtransaction", "gettxout",
                                 "getfeerate"};
 
+static void plugin_config_cb(const char *buffer,
```

Comment:
"We need our Bitcoin backend to be initialized, but the plugins have not yet been
started at this point."

(GH does not allow comments on commit msgs).

This seems like a can of worms.

OTOH, reordering to create the rpc before topology is ready is also problematic: there could innocently be an RPC call requiring topology information (plugin init, or anyone else) and the only sane semantic is to block at that point until we have it.

Can we create the rpc file, but not service it until we have topology?  That puts a burden on any `bitcoind` plugin not to deadlock, but it's simple?

Reply:
Ok, I had some edge cases when I tried but now it's working fine.

I'm still not sure about requiring the RPC socket to be initialized in `libplugin`, especially for plugins ""under"" `lightningd`.

Reply:
Hmmm I'll have to make the plugins initialization synchronous then

Reply:
> but not service it until we have topology

Ok I missed this part.